### PR TITLE
fix TypeError when create serial device

### PR DESCRIPTION
**Pull request by xutian**
ID: 1737744

Signed-off-by: Xu Tian <xutian@redhat.com>

### CODE REVIEW

File: virttest/qemu_vm.py [hunk 1]
```diff
@@ -1567,7 +1567,7 @@ def sort_key(dev):
             devices.insert(serial_devices)
 
             # Create virtio_ports (virtserialport and virtconsole)
-            serial_type = serial_params['serial_type']
+            serial_type = serial_params.get('serial_type', '')
```
Comment (vivianQizhu) [inline virttest/qemu_vm.py]: From my view `serial_type` must be defined when there is `serials` defined, do you have any use case that allow it is empty?

Comment (xutian) [reply virttest/qemu_vm.py]: if 'serial_type' is mandatory  param , line#2699 should be "if not params.object_params(serial)['serial_type']", base on your origin code, I guess serial_type is not a mandatory param

Comment (vivianQizhu) [reply virttest/qemu_vm.py]: @xutian Em, my fault, it should be mandatory, maybe you could help correct that:), thanks.

**Comment by yanan-fu** [comment]
The issue listed in 1737744 is `'NoneType' object has no attribute 'startwith'`,  i think this patch can not fix it,  with the new change, it should failed as `KeyError`.
Any other patch need work together with it ?

**Comment by xutian** [comment]
@yanan-fu 

"serial_type = None" because default value for  'get()' function of  dict is None. now, update code to serial_params['serial_type'] will raise  missing options exception before create serial device.

Thanks,
Xu

**Comment by xutian** [comment]
@yanan-fu  and @vivianQizhu  any more comments, if no let  me merge this small fix.

**Pull request merged**